collection-scripts/gather_services_status

```diff
@@ -77,6 +82,28 @@ get_cinder_status() {
     run_bg ${BASH_ALIASES[os]} volume backend pool list --long '>' "$CINDER_PATH"/pool_list
 }
 
+# Nova service gathering - sevices, hypervisors, cells, host mappings, allocation audit
+get_nova_status() {
+    local NOVA_PATH="$BASE_COLLECTION_PATH/ctlplane/nova"
+    mkdir -p "$NOVA_PATH"
+    run_bg ${BASH_ALIASES[os]} compute service list '>' "$NOVA_PATH"/service_list
+    run_bg ${BASH_ALIASES[os]} hypervisor list '>' "$NOVA_PATH"/hypervisor_list
+    run_bg /usr/bin/oc -n openstack exec -t nova-cell0-conductor-0 -- nova-manage cell_v2 list_cells '>' "$NOVA_PATH"/cell_list
+    run_bg /usr/bin/oc -n openstack exec -t nova-cell0-conductor-0 -- nova-manage cell_v2 list_hosts '>' "$NOVA_PATH"/host_list
+    run_bg /usr/bin/oc -n openstack exec -t nova-cell0-conductor-0 -- nova-manage placement audit --verbose '>' "$NOVA_PATH"/nova-manage_placement_audit
+    run_bg /usr/bin/oc -n openstack exec -t nova-cell0-conductor-0 -- nova-manage placement heal_allocations --dry-run --verbose '>' "$NOVA_PATH"/nova-manage_placement_heal_allocations
+}
+
+# Placement service gathering - capacity overview
+get_placement_status() {
+    local PLACEMENT_PATH="$BASE_COLLECTION_PATH/ctlplane/placement"
+    mkdir -p "$PLACEMENT_PATH"
+    # NOTE(gibi): this gives us a very simple resource capacity view of the
+    # cluster. It is intentionally uses 1 MB RAM query to get one candidate
+    # from each compute
+    run_bg ${BASH_ALIASES[os]} allocation candidate list --resource MEMORY_MB=1 --fit-width '>' "$PLACEMENT_PATH"/allocation_candidate_list
```

Comment:
in the other hand this should not be expensive as it runs a very similar query than any of the basic VM scheduling scenario

Reply:
this will not include any ironic nodes but yes it should be pretty quick.
however this could also get pretty large.

this will include all the provider summaries 

"provider_summaries": {
    "be99627d-e848-44ef-8341-683e2e557c58": {
      "resources": {},
      "traits": [
        "COMPUTE_VOLUME_MULTI_ATTACH"
      ],
      "parent_provider_uuid": null,
      "root_provider_uuid": "be99627d-e848-44ef-8341-683e2e557c58"
    },
    "9a9c6b0f-e8d1-4d16-b053-a2bfe8a76757": {
      "resources": {
        "VCPU": {
          "capacity": 4,
          "used": 0
        },
        "MEMORY_MB": {
          "capacity": 2048,
          "used": 0
        }
      },
      "traits": [
        "HW_NUMA_ROOT",
        "CUSTOM_FOO"
      ],
      "parent_provider_uuid": "be99627d-e848-44ef-8341-683e2e557c58",
      "root_provider_uuid": "be99627d-e848-44ef-8341-683e2e557c58"
    },
    "ba415f98-1960-4488-b2ed-4518b77eaa60": {
      "resources": {},
      "traits": [
        "CUSTOM_VNIC_TYPE_DIRECT"
      ],
      "parent_provider_uuid": "be99627d-e848-44ef-8341-683e2e557c58",
      "root_provider_uuid": "be99627d-e848-44ef-8341-683e2e557c58"
    },
    "92e971c9-777a-48bf-a181-a2ca1105c015": {
      "resources": {
        "NET_BW_EGR_KILOBIT_PER_SEC": {
          "capacity": 10000,
          "used": 0
        }
      },
      "traits": [
        "CUSTOM_PHYSNET1"
      ],
      "parent_provider_uuid": "ba415f98-1960-4488-b2ed-4518b77eaa60",
      "root_provider_uuid": "be99627d-e848-44ef-8341-683e2e557c58"
    },
    "cefbdf54-05a8-4db4-ad2b-d6729e5a4de8": {
      "resources": {
        "NET_BW_EGR_KILOBIT_PER_SEC": {
          "capacity": 20000,
          "used": 0
        }
      },
      "traits": [
        "CUSTOM_PHYSNET2"
      ],
      "parent_provider_uuid": "ba415f98-1960-4488-b2ed-4518b77eaa60",
      "root_provider_uuid": "be99627d-e848-44ef-8341-683e2e557c58"
    }

for each host.

that is useful just verbose.
that said its text data so it should be highly compressable so it likely wont increase the size of the result by much after its compressed. on a large cloud this still might be in the 10s of MBs

Reply:
what i would add is a list fo the resource classes and traits.

we can infer that form the provider summaries i guess but i think capturing them separately as well would be nice.

Reply:
Exactly the provider summary is the point of the call as it gives us the resource view of the cluster.

Reply:
we cant no,  since they always use a custom resource class we don't have generic way.
having said that we do plan to add some cloud wide capasity/usage enhancements to placement in the future so we may be able to support something like the owner triat to do that

the provider summaries is basically what the capsity API should return
so if we supported 

openstack provider summary show --trait OWNER:COMPUTE
openstack provider summary show --aggreate [<aggregate uuid>|all]

then i think we could address this in the future with that work.